pkg/kubelet/images/image_manager.go

```diff
@@ -64,27 +87,48 @@ func NewImageManager(recorder record.EventRecorder, imageService kubecontainer.I
 	} else {
 		puller = newParallelImagePuller(imageService, maxParallelImagePulls)
 	}
+	var pullImageSecretCheck bool
+	if pullImageSecretRecheck != nil {
```

Comment:
if we also check whether the feature gate is enabled here also, then we can always refer to pullImageSecretCheck instead of checking both

Reply:
Yes, True that. Addressed.